docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1700 @@
+{
```

Comment:
<p>Is this slow?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Reply:
Btw, `_prescrump` is parallelized now using all threads. Do you notice whether or not it is faster?

Reply:
I used the latest version of `_prescrump` (i.e. its parallelized version) and it was wayyy faster... If I remember correctly, the computation time was reduced by about 85% percent!! (from ~120 sec to ~18 sec)

---

**However, I think MERLIN is still slow in some cases** I recently tried to play with length of the `np.random.uniform` time series (input) and I tried to get `top-1` discord only. In one of cases (when `len(T)` is `50000` ), I noticed the `min_dist` reduced to a point where the algorithm found ~12000 candidates with input `right=True` (in function `_find_candidates`) and then, with `right=False`, the  number of final candidates becomes ~6000. The running time to find top-1 was about 85 minutes! (I used `decay=0.05` instead of `0.01`.)
 
Is there any function in STUMPY that can give the distance matrix between subsequences of `T_A` and subsequences of `T_B`? (Because, I think iterating through the subsequences one by one does not take advantage of the overlap between `T[i:i+m]` and `T[i+1:i+1+m]`... So, maybe we should do iteration block by block (similar to MERLIN3)..that might help...

Reply:
<p>Actually, after we finish looking at the <code>discords</code>  function,  I think it is a good place to pause any work that improves the computational time and, instead, switch our focus to:</p><p> </p><ol><li>Move the functions into <code>stumpy/discords.py</code> </li><li>Write unit tests to cover everything</li><li>Update this tutorial to call the <code>discords</code> function</li></ol><p> </p><p>Once all of the unit tests are complete, then that is the safest/best time to think about optimizations. We want to make sure that our code is the easiest to understand and write unit tests for first. Afterward, we can be certain that our optimizations didn't break anything as long as our unit tests all pass. Does that make sense? If we perform optimizations now, it will be hard to debug if anything goes wrong and we may not catch things that go wrong.</p>

Reply:
<p>Again, before we do the steps above, we should complete the <code>discords</code> function first (along with the comments above regarding <code>min_dist</code>. We are SO close! Can you see if that function is ready for review? </p>

Reply:
You are right... before making function more complicated, we should go for the unit test to discover any potential bugs. Thanks for the input. I will checkout the top-level function `discord`. I think I might need to revise it a little  bit as I have not touched it for a while. I will let you know soon (in a day or two). And, I will think about `min_dist` (and other comments).

Reply:
<p>Great! Honestly, the code above is looking beautiful and you have a lot to be proud of. There is still a lot more to be done but this is a really solid foundation that we are building on top of. </p>